src/js/calculator.js

```diff
@@ -0,0 +1,107 @@
+import { divideOperatorAndNumber } from '../utils/index.js';
+
+class Calculator {
+  constructor({ $target }) {
+    this.$target = $target;
+    this.sum = '';
+    this.operator = '';
+    this.total = document.getElementById('total');
+    this.initialize();
+  }
+
+  add(a, b) {
+    this.sum = Number(a) + Number(b);
+    this.renderTotal();
+  }
+
+  subtract(a, b) {
+    this.sum = a - b;
+    this.renderTotal();
+  }
+
+  multiple(a, b) {
+    this.sum = a * b;
+    this.renderTotal();
+  }
+
+  divide(a, b) {
+    this.sum = Math.round(a / b);
+    this.renderTotal();
+  }
+
+  clear() {
+    this.sum = '0';
+    this.operator = '';
+    this.renderTotal();
+  }
+
+  renderTotal() {
+    this.total.innerText = this.sum;
+  }
+
+  makeResult() {
+    const [a, b, operator] = divideOperatorAndNumber(this.sum);
+    if (operator === '+') return this.add(a, b);
+    if (operator === '-') return this.subtract(a, b);
+    if (operator === 'X') return this.multiple(a, b);
+    if (operator === '/') return this.divide(a, b);
+  }
+
+  addNumberListener() {
+    const numberButtons = document.querySelectorAll('.digit');
+    numberButtons.forEach((element) => {
+      element.addEventListener('click', (e) => {
+        //숫자가 3자리수 이상 기입되는 경우 경고를 해준다.
+        const [a, b, _] = divideOperatorAndNumber(
+          this.sum + e.target.innerText
+        );
```

Comment:
```suggestion
        const [a, b] = divideOperatorAndNumber(
          this.sum + e.target.innerText
        );
```

사용하지 않을 땐 받지 않아도 됩니다.